### ISSUE TITLE
Thread.startVirtualThread does not propogate context

### ISSUE THREAD

**Issue by anuragagarwal561994** [open-telemetry/opentelemetry-java-instrumentation#11950]
### Describe the bug

When a virtual thread is started context should propogate from the thread starting the virtual thread, but this is not happening instead a new context is started.

### Steps to reproduce

`Thread.startVirtualThread(() -> log.info("span"))` this will initiate a new context and not log the respective trace id originated from the parent context

The same is also application for methods such as:
```
Thread.ofPlatform().start(() -> log.info("span"))
Thread.ofVirtual().start(() -> log.info("span"))
```
 

### Expected behavior

Context propagation should happen and the virtual thread should have the same context as the thread initiating the start.

### Actual behavior

Context propagation does not take place

### Javaagent or library instrumentation version

2.5.0

### Environment

**JDK**: Temurin-21+35 (build 21+35-LTS)
**OS**: MacOs

**Comment by heyams** [comment]
Please check out this doc https://opentelemetry.io/docs/languages/java/instrumentation/#context-propagation-between-threads.
Hope that helps.

**Comment by anuragagarwal561994** [comment]
No this I know, just wanted to understand why the executors instrumentation doesn't handle this case.

**Comment by trask** [comment]
Related to #9534

@JonasKunz wdyt?

**Comment by anuragagarwal561994** [comment]
That is a very good explanation to understand the issue, the thing is in some cases you would want to do things in background in virtual threads, but thats short lived thing. So for example I have an entity in my request which I would want to cache asyncronously in my distributed cache so I can just do `Thread.startVirtualThread` and won't join it as with timeouts and everything configured in the actual sync call inside of the runnable I can assume that it would get complete. In such cases may be it would be beneficial for us to have context propogation

**Comment by JonasKunz** [comment]
I agree that there are cases / preferences where virtual threads might be directly started. That's why I think it could be useful to have the instrumentation, but have it opt-in. It might also be difficult to get that instrumentation to not interfer with the existing executor instrumentations, so it could be more work to implement than it initially seems.

**Comment by trask** [comment]
Since there has been no activity on this enhancement for the past year we are closing it to help maintain our backlog. Anyone who would like to work on it is still welcome to do so, and we can re-open it at that time.
